packages/web3-eth-contract/src/types.ts

```diff
@@ -34,40 +42,40 @@ export interface ContractOptions {
 	readonly from?: Address;
 	readonly data?: Bytes;
 	jsonInterface: ContractAbi;
-	address?: Address | null;
+	address?: Address;
 }
 
 export interface ContractInitOptions {
-	readonly gas: Uint | null;
-	readonly gasPrice: Uint | null;
+	readonly gas?: Uint | null;
+	readonly gasPrice?: Uint | null;
```

Comment:
Should also support EIP1559 type params, or create TODO new issue

Reply:
https://github.com/ChainSafe/web3.js/issues/4685